src/AWS.Deploy.Orchestration/RecommendationEngine/RecommendationEngine.cs

```diff
@@ -17,6 +17,7 @@ public class RecommendationEngine
     {
         private readonly IList<RecipeDefinition> _availableRecommendations = new List<RecipeDefinition>();
         private readonly OrchestratorSession _orchestratorSession;
+        private readonly HashSet<string> _uniqueRecipeId = new HashSet<string>();
```

Comment:
I don't see `_uniqueRecipeId` being used other then adding items to it.

Reply:
Fixed. changed it to a local variable